google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java

```diff
@@ -1133,6 +1136,9 @@ public Builder enableGrpcGcpExtension() {
     public Builder enableGrpcGcpExtension(GcpManagedChannelOptions options) {
       this.grpcGcpExtensionEnabled = true;
       this.grpcGcpOptions = options;
+      if (this.numChannels == null) {
+        this.numChannels = GRPC_GCP_ENABLED_DEFAULT_CHANNELS;
+      }
```

Comment:
Remove this and instead assign the default number of channels in the build method for both GCP enabled and disabled. (See also below)